torch/csrc/jit/passes/replacement_of_old_operators.cpp

```diff
@@ -0,0 +1,110 @@
+#include <torch/csrc/jit/passes/replacement_of_old_operators.h>
+
+#include <c10/util/Exception.h>
+#include <caffe2/serialize/versions.h>
+#include <torch/csrc/jit/frontend/schema_matching.h>
+#include <torch/csrc/jit/ir/irparser.h>
+#include <torch/csrc/jit/operator_upgraders/upgraders.h>
+#include <torch/csrc/jit/operator_upgraders/utils.h>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+#include <torch/csrc/jit/runtime/graph_iterator.h>
+#include <limits>
+#include <string>
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+
+static std::unordered_map<std::string, std::shared_ptr<Graph>> upgraderCache;
+
+std::shared_ptr<Graph> getUpgraderGraph(const std::string& upgrader_name) {
+  auto it = upgraderCache.find(upgrader_name);
+  if (it != upgraderCache.end()) {
+    return it->second;
+  }
+
+  auto upgrader_graph_entry = dump_upgraders_map().find(upgrader_name);
+  TORCH_INTERNAL_ASSERT(
+      upgrader_graph_entry != dump_upgraders_map().end(),
+      "Corresponding upgrader graph for ",
```

Comment:
Users might not be aware of what an upgrader is. Can we frame it as "Op X at older version Y is out of support window, please regenerate your model with newer version of PyTorch"

Reply:
Same for the warning on line 66